src/main/java/org/commcare/formplayer/auth/HmacAuthFilter.java

```diff
@@ -108,19 +108,23 @@ private HqUserDetailsBean getUserDetails(HttpServletRequest request) {
         JSONObject body = RequestUtils.getPostData(request);
         if (body.has("username") && body.has("domain")) {
```

Comment:
@snopoke I am not sure what's the right course of action here. Should we be checking for feature flags in the response body and setting them here ? I am not sure what requests end up here and whethery they will have the `enabled_previews` and `enabled_toggles` specified.

Reply:
this path won't currently have any FF set since HQ isn't sending them through with these requests. I suppose it should be sending them so let's add a check and if they are present we can use them otherwise use an empty list.